consul-lambda/client/ssm.go

```diff
@@ -52,14 +52,21 @@ func (c *SSMClient) Get(ctx context.Context, key string) (string, error) {
 // Set writes the value for the given key to Parameter Store.
 // It writes the value as an encrypted SecureString.
 // Any existing data for the given key is overwritten.
-func (c *SSMClient) Set(ctx context.Context, key, val string) error {
+func (c *SSMClient) Set(ctx context.Context, key, val string, advancedTier bool) error {
+	tierType := types.ParameterTierStandard
+
+	if advancedTier {
+		tierType = types.ParameterTierAdvanced
+	}
+
 	_, err := c.client.PutParameter(
 		ctx,
 		&ssm.PutParameterInput{
 			Name:      &key,
 			Value:     &val,
 			Overwrite: true,
 			Type:      types.ParameterTypeSecureString,
+			Tier:      tierType,
 		},
 	)
```

Comment:
So I'm currently thinking that maybe it would be better to determine the  byte size of `val` prior and if it exceeds standard tier limit (or if we get the size too large error back from AWS) then we move to advanced if the env var is set to true and the parameter needs to be.

That way if someone has parameters that *can* fit into a standard tier we put them there that way, to keep cost in mind (even though its not expensive, its still an added cost if we just add a blanket config to put everything in advanced tier).

Thoughts?